packages/core/content-manager/server/controllers/relations.js

```diff
@@ -65,6 +65,15 @@ module.exports = {
           return ctx.forbidden();
         }
       }
+    } else {
+      // eslint-disable-next-line no-lonely-if
+      if (entityId) {
+        const entity = await strapi.entityService.findOne(model, entityId);
+
+        if (!entity) {
+          return ctx.notFound();
+        }
+      }
     }
```

Comment:
Was looking at the no-lonely-if rule, why not doing this?
```suggestion
    } else if (entityID) {
        const entity = await strapi.entityService.findOne(model, entityId);
        if (!entity) {
          return ctx.notFound();
        }
    }
```

Reply:
Because I want it explicit that either we are dealing with a component or we are not. With ESlint logic here I find it less readable. We may have more code around the `if (entityId) {` for RBAC checks, like above, but for now we don't.